src/EnduranceTests/LayoutRenderer/AggregateExceptionLayoutRenderer.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace NServiceBus.AzureServiceBus.EnduranceTests.LayoutRenderer
+{
+    using System;
+    using System.Linq;
+    using System.Text;
+    using NLog;
+    using NLog.Config;
+    using NLog.LayoutRenderers;
+
+    /// <summary>
+    /// Exception information provided through
+    /// a call to one of the Logger.*Exception() methods.
+    /// </summary>
+    [LayoutRenderer("aggregateexception")]
+    [ThreadAgnostic]
+    public class AggregateExceptionLayoutRenderer : ExceptionLayoutRenderer
+    {
+        protected override void Append(StringBuilder builder, LogEventInfo logEvent)
+        {
+            base.Append(builder, logEvent);
+
+            var exception = logEvent.Exception as AggregateException;
+
+            if (exception == null)
+            {
+                return;
+            }
+
+            builder.AppendLine(string.Empty);
+            builder.AppendLine(string.Empty);
+            builder.AppendLine("AggregateException Inner Exceptions:");
+
+            foreach (var innerLogEvent in exception.InnerExceptions.Select(iex => new LogEventInfo
```

Comment:
`exception` at this point is an `AggregateException`. Better to iterate over `exception.Flatten().InnerExceptions` to get the real exceptions and not child `AggregateException`.